web/src/pm_conversations.ts

```diff
@@ -90,6 +95,48 @@ class RecentDirectMessages {
             .map((conversation) => conversation.user_ids_string);
     }
 
+    has_conversation(user_ids_string: string): boolean | undefined {
+        const recipient_ids_string = util.extract_pm_recipient_ids_string(user_ids_string);
+        // Check if there are any previous messages in the conversation.
+        const previous_direct_messages_exist = all_messages_data
+            .all_messages()
+            .find((message) => message.is_private && message.to_user_ids === recipient_ids_string);
```

Comment:
Why does this involve looping through `all_messages`, which could process 10,000s of messages? 

`recent_message_ids` is a map from the `user_ids_string` to the data we need...

Reply:
Fixed that now.